src/components/Form.js

```diff
@@ -159,30 +159,38 @@ class Form extends React.Component {
             const defaultValue = this.props.draftValues[inputID] || child.props.defaultValue;
 
             // We want to initialize the input value if it's undefined
-            if (_.isUndefined(this.inputValues[inputID])) {
-                this.inputValues[inputID] = defaultValue;
+            if (_.isUndefined(this.state.inputValues[inputID])) {
+                this.state.inputValues[inputID] = defaultValue;
+            }
+
+            if (!_.isUndefined(child.props.value)) {
+                this.state.inputValues[inputID] = child.props.value;
```

Comment:
I just bumped into this code and noticed that we are mutating the state during rendering. I haven't seen any bug caused by this, but, as far as I know, a base assumption when working in React that the state won't mutate like this, otherwise, basic checks equality checks by reference become not reliable.

Reply:
Is there any strong reason to mutate the state in this case?

Reply:
I think you're right that this shouldn't be done. I think the state will change, but we won't re-render. AFAIK we should always be using `setState` cc @rushatgabhane

Reply:
It seems like this can be a technique to exactly do that, avoid re-rendering. In some cases it makes sense to avoid re-rendering.. but there are other ways of achieving this without having to mutate the state. There was a similar conversation here about a proposal suggesting to do this: https://github.com/Expensify/App/issues/11086#issuecomment-1323026474

Reply:
> > To have a "state" you can change without re-rendering, I think the recommended way is:
> > Class component: member variable
> 
> Agree. But why maintain another duplicate variable that we'll need to always keep in sync with `inputValues`?

I haven't really gone in depth in this code, and I don't really plan to at the moment because of priorities, but even in the worse case of having duplicate variables, I still think it is better that doing something that is not expected in the framework.

Reply:
Awesome, thanks guys!